### PR TITLE
Improve Binary String search performance

### PR DESCRIPTION
**Pull request by AaronOpfer**
This made finding a binary constant in my debuggee take about 5-10 seconds instead of not being complete after several minutes.

The main issue was the pointless usage of `ByteShiftArray` when we already had a contiguous view of the region memory. Constantly popping bytes into and out of the BSA destroyed all performance in my profiler (80% of the runtime). Calls to `region->size()` were second on the list (13% of runtime) because we called it inside the while loop.

### CODE REVIEW

File: plugins/BinarySearcher/DialogBinaryString.cpp [hunk 1]
```diff
@@ -101,7 +93,10 @@ void DialogBinaryString::do_find() {
 						}
 					}
 
-					ui->progressBar->setValue(util::percentage(i, regions.size(), p - &pages[0], region->size()));
+					// update progress bar every 64KB
+					if (quint64(p) & 0xFFFF == 0) {
```
Comment (10110111) [inline plugins/BinarySearcher/DialogBinaryString.cpp]: This condition is never met.

Comment (AaronOpfer) [inline plugins/BinarySearcher/DialogBinaryString.cpp]: I'm having trouble figuring out how to add compiler flags to a cmake project.

Comment (10110111) [inline plugins/BinarySearcher/DialogBinaryString.cpp]: `cmake -DCMAKE_CXX_FLAGS=your-flags`

Comment (AaronOpfer) [inline plugins/BinarySearcher/DialogBinaryString.cpp]: Okay, thanks. I'll make another PR addressing this and some other pedantic warnings I've littered in my PRs.

**Pull request merged**